### ISSUE TITLE
Mac OSX builds failing

### ISSUE THREAD

**Issue by demorest** [conda-forge/tempo2-feedstock#36]
### Solution to issue cannot be found in the documentation.

- [x] I checked the documentation.

### Issue

Mac builds are failing to compile the C++ code.  Some more info is in PR #35

**Comment by demorest** [comment]
The TLDR is that we were using a too-old version of the C++ compiler.   We had specified version 15 in conda_build_config.yaml, does anyone know why this was hard-coded rather than letting it go with the current conda-forge defaults?  I've bumped it up to 17 which has helped though there are now some other errors with at least one of the plugins (I think these are fixable, will get to that shortly).

More info:

The original error was failure to find basic C++ header files like `<string>`.  This was due to changes discussed at https://github.com/conda-forge/clangdev-feedstock/issues/310 where the C++ header files were split into a different package (presumably for reasons).  However this change was only backported to clang>=16.

I first tried adding the libcxx-devel package to get the headers.  This addressed the header issue but the build failed with other obscure C++ errors, including a warning that this libcxx version requires clang 17. Increasing version to 17 fixed these errors.

The current problem is that the "interpolate" function defines a `sample` struct which has a name collision with `std::sample`.  I'll try to poke at this offline for a bit, if I can reproduce the error here.
